### PR TITLE
NODE-2257: add server description equality checks

### PR DESCRIPTION
**Pull request by mbroadst**
## Description
`ServerDescription` gains an `equals` method, which allows us to compare against another description for equality. The new operator is used in the `Topology` update path in order to debounce topology update events when nothing of note has actually changed.

Additionally, a tool is introduced with this PR called `sdam_viz` which allows us see changes to a cluster in realtime. It's in a very nascent state, but it's very useful for watching topology changes

**NOTE:** I'd prefer not get into review of the `sdam_viz` tool, we can improve that incrementally.

### CODE REVIEW

File: lib/core/utils.js [hunk 1]
```diff
@@ -163,6 +163,10 @@ function isUnifiedTopology(topology) {
   return topology.description != null;
 }
 
+function arrayStrictEqual(arr, arr2) {
```
Comment (daprahamian) [inline lib/core/utils.js]: Are we ever going to have a case where the arrays we are using this on do not exist? Should we add an `Array.isArray(arr)`?

File: test/tools/sdam_viz [hunk 2]
```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env node
```
Comment (daprahamian) [inline test/tools/sdam_viz]: neat!

File: lib/core/sdam/server_description.js [hunk 3]
```diff
@@ -119,6 +121,30 @@ class ServerDescription {
...
   }
+
+  /**
+   * Determines if another `ServerDescription` is equal to this one per the rules defined
+   * in the {@link https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#serverdescription|SDAM spec}
+   *
+   * @param {ServerDescription} other
+   * @return {Boolean}
+   */
+  equals(other) {
```
Comment (daprahamian) [inline lib/core/sdam/server_description.js]: any chance of `other` being `undefined` or `null`?

Comment (mbroadst) [reply lib/core/sdam/server_description.js]: hm good call, I'll add in a check for that

File: lib/core/utils.js (hunk 1 above)
Comment (mbroadst) [reply lib/core/utils.js]: added

**Pull request merged**